python/ray/autoscaler/autoscaler.py

```diff
@@ -439,13 +441,9 @@ def can_update(self, node_id):
             return False
         return True
 
-    def launch_new_node(self, count, instance_type=None):
+    def launch_new_node(self, count, instance_type):
```

Comment:
Can we `assert instance_type is not None` here?

Reply:
nvm `instance_type` is None here if we're not using the demand scheduler. 

Can we leave a comment/docstring here? It's a little unintuitive/hard to keep track of the fact that `instance_type` is sometimes None, but sometimes defined in the default value case